tests/schema-builder.spec.ts

```diff
@@ -0,0 +1,345 @@
+            });
+
+            it('should build schema for object', async () => {
+                const schema = generateSchema({});
+                expect(schema).toEqual({ type: 'object' });
+            });
+        });
+
+        describe('arrays', () => {
+            it('it should generate schema for arrays of the same type', async () => {
+                const schema = generateSchema([1, 2, 3]);
+                expect(schema).toEqual({ type: 'array', items: { type: 'integer' } });
+            });
+
+            it('it should generate schema for arrays of the same type with floats', async () => {
+                const schema = generateSchema([1, 2.1, 3]);
+                expect(schema).toEqual({ type: 'array', items: { type: 'number' } });
+            });
+
+            it('it should generate schema for arrays of different primitive types', async () => {
+                const schema = generateSchema([1, 1.1, 'string', null, false, true]);
+                expect(schema).toEqual({ type: 'array', items: { type: ['null', 'boolean', 'number', 'string'] } });
+            });
+
+            it('it should generate schema for arrays of different primitive types and ints only', async () => {
+                const schema = generateSchema([1, 'string', null, false, true]);
+                expect(schema).toEqual({ type: 'array', items: { type: ['null', 'boolean', 'integer', 'string'] } });
+            });
+        });
+
+        describe('objects', () => {
+            it('it should generate schema for object with props of the same type', async () => {
+                const schema = generateSchema({ one: 1, two: 2 });
+                expect(schema).toEqual({
+                    type: 'object',
+                    properties: { one: { type: 'integer' }, two: { type: 'integer' } },
+                    required: ['one', 'two'],
+                });
+            });
+
+            it('it should generate schema for object with props of different types', async () => {
+                const schema = generateSchema({ one: 1, two: 'second' });
+                expect(schema).toEqual({
+                    type: 'object',
+                    properties: { one: { type: 'integer' }, two: { type: 'string' } },
+                    required: ['one', 'two'],
+                });
+            });
+
+            it('it should generate schema for object with props of different types w/o required', async () => {
+                const schema = generateSchema({ one: 1, two: 'second' }, { noRequired: true });
+                expect(schema).toEqual({
+                    type: 'object',
+                    properties: { one: { type: 'integer' }, two: { type: 'string' } },
+                });
+            });
+        });
+
+        describe('nested array', () => {
+            it('should generate schema for nested arrays', async () => {
+                const schema = generateSchema([1, [2.1], [[3]]]);
+                expect(schema).toEqual({
```

Comment:
maybe extract big expected payloads to different files? Will make it easier to follow to file

Reply:
extracted to snaphots and files.